### ISSUE TITLE
Cluster config appears different in UI

### ISSUE THREAD

**Issue by elibola** [rancher/terraform-provider-rancher2#1050]
Hello,

When we create a cluster with terraform rancher2 provider, it takes configurational parameters properly. But when we try to view or edit the cluster config on UI, parameters shown are different than what we assigned. For instance; datastore variable is set to "PURE_DatastoreGP26", and VMs are located in this datastore, but on UI when editing cluster config, datastore info doesn't seem as we set, so when we change something else in config (UI) and save the configuration, all the cluster config changes to the undesired set of values. Not only datastore, also folder, vm template and other parameters seem different on UI.

```
resource "rancher2_machine_config_v2" "machine_config_worker" {
  generate_name = "vsphere-config"

  vsphere_config {
    creation_type = "template"
    clone_from    = "rhel8-rke2-template"
    cpu_count   = var.worker_node_cpu_count
    memory_size = var.worker_node_memmory
    disk_size   = var.worker_node_disksize
    datacenter  = var.datacenter
    datastore   = var.datastore_worker
```

```
variable "datastore_worker" {
  type = string
  default = "PURE_DatastoreGP26"
}
```

![image](https://user-images.githubusercontent.com/72225896/210211490-2e3c4199-6f4c-4869-a718-71c42aaf5da8.png)

Rancher v2.6.9
rancher2 provider version 1.24.1

**Comment by nickvth** [comment]
yes folder is also empty, but set when provisioning via terraform.

**Comment by GaboFDC** [comment]
You need to specify the full path, sometimes the UI doesn't show it, but API does, so You need to put the whole thing in tf, for the UI to show it properly.
i.e something like `datastore = "/<datacenter>/datastore/<Folder>/PURE_DatastoreGP26"`

**Comment by matttrach** [comment]
This issue is a few years old now and only references unsupported Rancher and provider versions, so I am going to close it for now, but feel free to open a new issue if this is still relevant.

It is generally not expected for the terraform provider and the UI to work similarly or interchangeably, but this is a good goal for #1940.
